### DISCUSSION TITLE
D086-ZG only connects one channel

### DISCUSSION THREAD

**Discussion by dunderklumpen-swe** [Koenkk/zigbee2mqtt#13622] [Devices]
Hi! I installed this dual dimmer today, and pairing works fine on one channel. Since it's dual, it should show two separate channels, but in "Exposes" it only shows one state and one brightness. Shouldn't it be "L1 endpoint" and "L2 endpoint"?

How do I go about getting the other channel connected?

Setup: zigbee2mqtt and mosquitto in home assistant on a nuc. Zigbee-dongle: Zigbee 3.0 USB dongle, Sonoff ZBDongle-P

Thanks in advance!

**Comment by Koenkk** [comment]
@andredahlqvist  can you check this?

**Reply by andredahlqvist** [reply]
@Koenkk Sure, I'll try my best. I only use one of the channels on my device, that's why I didn't notice.

**Reply by dunderklumpen-swe** [reply]
Thanks a lot, André! If you need any testing done, just let me know. I have both channels connected to both switch and lights.

**Reply by dunderklumpen-swe** [reply]
@andredahlqvist I’m new to discussions and changes on GitHub, so I’m not sure how things are done here. My apologies if I’m not posting this correctly. I would just like to know when I can expect this issue to be resolved? Is there something I could or should do, like create an issue?

Thanks in advance!

**Reply by andredahlqvist** [reply]
I've just contributed support for this device since I happened to own it. Its my first contribution to this project, so I'm not really very involved with the development. I will try to find the time to look into this.

**Reply by andredahlqvist** [reply]
I've fixed this in the below Pull request:

https://github.com/Koenkk/zigbee-herdsman-converters/pull/4691

**Reply by dunderklumpen-swe** [reply]
> I've fixed this in the below Pull request:
> 
> [Koenkk/zigbee-herdsman-converters#4691](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4691)

Thank you very much, André!

**Reply by dunderklumpen-swe** [reply]
I have confirmed now that this works. Didn't even have to re-pair the device, it just started working. Stunning work! Thank you very much, @andredahlqvist and @Koenkk!

**Reply by andredahlqvist** [reply]
@dunderklumpen-swe I noticed that if you change the dimming level on the physical switch it didn't update correctly in Home Assistant. I'm pretty sure I broke this with my latest change, since it works if I revert it. Will investigate.

**Reply by dunderklumpen-swe** [reply]
Ok. The first test I did was to just turn on and change brightness from HA, and that worked. Now I tested the physical switch, and you are correct: it doesn't update in HA. If I turn the light on with physical switch it still says "off" in HA.

At least now it works for me to turn off all lights through a script, so that's nice. Hopefully you'll be able to fix it for us. Thanks!

**Reply by dunderklumpen-swe** [reply]
@andredahlqvist Have you had any chance to investigate why the values doesn't show in HA? It still works to "turn all lights off", so that's good, but it would be nice to have the correct values anyway.

**Reply by andredahlqvist** [reply]
No. I've not had time to look at it. And I have a hard time to find a good example to use as base. @Koenkk Do you know of any existing supported dimmer with two channels that I can look at for clues?

**Reply by andredahlqvist** [reply]
@dunderklumpen-swe I think I have a version that supports both channels and where changes via the physical switch is reported to Zigbee. Can you e-mail me and I'll send you the file to test? My e-mail is my username in github, plus gmail.com.

**Reply by dunderklumpen-swe** [reply]
@andredahlqvist That's really good news! I'll send you an email right away!

**Reply by albatorsk** [reply]
Eagerly awaiting the state reporting fix. Will it be included in the next Zigbee2MQTT release? If not, I will gladly help testing it as I have the same D086-ZG as @dunderklumpen-swe

**Reply by dunderklumpen-swe** [reply]
@andredahlqvist  and/or @albatorsk  Has there been any new progress made on this? It's really only a minor inconvinience right now, but I'm thinking about putting a NSpanel or similar in my house, and then correct values gets a bit more important.

Also, let me know if I can assist in testing, somehow. All the best!

**Reply by albatorsk** [reply]
@dunderklumpen-swe Did you email @andredahlqvist for the files to test his potential fix? I did not, as I did not want to send an unsolicited mail.

**Reply by andredahlqvist** [reply]
I installed this one behind a dual switch in the living room, so now I have more incentive to actually fix this:-). I'll see what I can do. It's pretty much guesswork for me though, because I can't find another supported dimmer with two channels that I can copy from.

**Reply by dunderklumpen-swe** [reply]
> @dunderklumpen-swe Did you email @andredahlqvist for the files to test his potential fix? I did not, as I did not want to send an unsolicited mail.
@albatorsk Yes I did. But unfortunately there was no improvment from the first one that's implemented in the repo.


> I installed this one behind a dual switch in the living room, so now I have more incentive to actually fix this:-). I'll see what I can do. It's pretty much guesswork for me though, because I can't find another supported dimmer with two channels that I can copy from.
@andredahlqvist Thank you very much for your efforts! If I can assist with testing I will gladly do that. Unfortunately, my own expertise in coding is limited to "copy/paste from someone on a internet forum" or "I will follow this tutorial to the letter"... But, as stated, I will gladly help in any way I can!

**Reply by franssonjohan** [reply]
Would really appreciate a state reporting fix here too. Hopefully, you will manage to solve it. Thanks in advance!

**Reply by andredahlqvist** [reply]
A fix for this was merged last week. It's not perfect yet when dimming up and down (fluctuates a bit), but pretty close.

https://github.com/Koenkk/zigbee-herdsman-converters/pull/5410

**Reply by franssonjohan** [reply]
Thanks! Will give it a try.

**Reply by franssonjohan** [reply]
@andredahlqvist Works well now! Thank you
